### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by christian-schilling**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,79 +2,118 @@
...
+
+With everything stored in one repo, CI/CD system only need to look into one source for each particular
+deliverable.
+However, building multiple deliverables from from one big tree, introduces a new complexity
+into the build system: Now build tools or package managers, typically taylored to specfic languages
+or toolchains, need to be used to answer the question: "What deliverables are affected by a given commit
+and need to be rebuild?".
+
+With Josh, each deliverable gets it's own virtual git repository with dependencies declared in the `workspace.josh`
+file. This means answering above question becomes as simple as comparing commit ids.
```
Comment (LMG) [inline README.md:89]: the above question

File: README.md [hunk 2]
```diff
@@ -2,79 +2,118 @@
...
+```
+
+### Simplfied CI/CD
+
+With everything stored in one repo, CI/CD system only need to look into one source for each particular
+deliverable.
+However, building multiple deliverables from from one big tree, introduces a new complexity
+into the build system: Now build tools or package managers, typically taylored to specfic languages
+or toolchains, need to be used to answer the question: "What deliverables are affected by a given commit
+and need to be rebuild?".
```
Comment (LMG) [inline README.md:86]: this paragraph is a bit confusing

File: README.md [hunk 3]
```diff
@@ -2,79 +2,118 @@
...
+
+With Josh, each deliverable gets it's own virtual git repository with dependencies declared in the `workspace.josh`
+file. This means answering above question becomes as simple as comparing commit ids.
+Furthermore due to the tree filtering each build is guaranteed to be perfectly sandboxed
+and only sees those parts of the monorepo that have actually been mapped.
+
+This also means the deliverables to be re-build can be selected without cloning any repos like
```
Comment (LMG) [inline README.md:93]: selected?

File: README.md [hunk 4]
```diff
@@ -2,79 +2,118 @@
...
+typically necessary with normal build tools.
+
+### GraphQL API
+
+It is often desireable to access content stored in git without requireing a clone of the repository.
```
Comment (LMG) [inline README.md:98]: requiering

File: README.md [hunk 5]
```diff
@@ -2,79 +2,118 @@
...
+This is usefull for CI/CD systems or web frontends like dashboards.
```
Comment (LMG) [inline README.md:99]: useful

File: README.md [hunk 6]
```diff
@@ -2,79 +2,118 @@
...
+  }
+}
+```
+
+
+### Caching proxy
+
+Even without using the more advanced features like partial cloning or workspace,
+`josh-proxy` can act as a cache to reduce traffic between locations or keep your CI from
+doing lot's of requests to the main git host.
```
Comment (LMG) [inline README.md:117]: lots

File: README.md (hunk 3 above)
Comment (robw-23) [inline README.md:93]: re-build -> rebuilt
any repos -> any additional repos

File: README.md (hunk 5 above)
Comment (robw-23) [inline README.md:99]: like -> such as

File: README.md [hunk 7]
```diff
@@ -2,79 +2,118 @@
...
+
+This also means the deliverables to be re-build can be selected without cloning any repos like
+typically necessary with normal build tools.
+
+### GraphQL API
+
+It is often desireable to access content stored in git without requireing a clone of the repository.
+This is usefull for CI/CD systems or web frontends like dashboards.
+
+Josh exposes a GraphQL API for that purpose. For example it can be used to find all workspaces currently
```
Comment (robw-23) [inline README.md:101]: Comma after "for example"

File: README.md [hunk 8]
```diff
@@ -2,79 +2,118 @@
...
+  rev(at:"refs/heads/master", filter:"::**/workspace.josh") {
+    files { path }
+  }
+}
+```
+
+
+### Caching proxy
+
+Even without using the more advanced features like partial cloning or workspace,
```
Comment (robw-23) [inline README.md:115]: workspace -> workspaces

File: README.md (hunk 6 above)
Comment (robw-23) [inline README.md:117]: doing -> performing
I'd rather you chose "many" over "lots of", so long as you also change "doing" to "performing"

File: README.md (hunk 3 above)
Comment (christian-schilling) [reply README.md]: not additional. I mean you literatlly need to clone zero repositories to answer that question.

**Pull request merged**